src/main.ts

```diff
@@ -199,15 +203,19 @@ async function injectViteIndexMiddleware(
   app.get("/*", async (req, res, next) => {
     if (isIgnoredPath(req.path, req)) return next();
 
-    const template = fs.readFileSync(
-      path.resolve(config.root, "index.html"),
-      "utf8"
-    );
-
     if (isStaticFilePath(req.path)) next();
     else {
-      const html = await server.transformIndexHtml(req.originalUrl, template);
-      res.send(getTransformedHTML(html, req));
+      try {
+        const basePath = getBasePath(req.path);
+        const template = fs.readFileSync(
+          path.join(config.root, basePath, "index.html"),
+          "utf8"
+        );
+        const html = await server.transformIndexHtml(req.originalUrl, template);
+        res.send(getTransformedHTML(html, req));
+      } catch (e) {
+        res.sendStatus(404);
```

Comment:
When looking for `index.html` file to serve will fail, you can just let it throw an 500 error, so don't try-catch this block at all.

Alternatively, you can invoke `next()` and pass an `e` error object as a parameter, that way you give an opportunity for other handlers to do it's job. For example, someone could have a middleware that serves beautiful 404 page when that error comes, but in your case you would make that impossible by closing the request too early.

See [this](https://expressjs.com/en/guide/using-middleware.html#middleware.error-handling) for more information.